contracts/core/Dispatcher.sol

```diff
@@ -578,6 +594,22 @@ contract Dispatcher is OwnableUpgradeable, UUPSUpgradeable, IDispatcher {
         isMatch = Ibc._hexStrToAddress(portId[portPrefixLen:]) == addr;
     }
 
+    function _addNewConnection(string calldata connection, LightClient lightClient) internal {
+        if (address(lightClient) == address(0)) {
+            revert IBCErrors.invalidAddress();
+        }
+        if (bytes(connection).length == 0) {
+            revert IBCErrors.invalidConnection("");
+        }
+        if (_connectionToClientId[connection] != 0) {
+            revert IBCErrors.invalidConnection(connection);
+        }
+
+        uint256 newClientId = ++_numClients; // Cache clientId to save SLOAD call
+        _connectionToClientId[connection] = newClientId;
+        _clientIdToLightClient[newClientId] = lightClient;
```

Comment:
can we directly map keccak(connectionID) -> lightClient?

Reply:
Actually now that I think about it again, I think we can directly map connectionId -> lightClient, and we can get rid of the (clientID) altogether. Though I'm curious why you are suggesting to hash the conneciton id in the mapping rather than storing it directly; is that a potential gas optimisation? 

I had clientId originally as a way to "white list" what light clients can be updated in `updateClientWithOptimisticConsensusState` to avoid calling `updateClientWithOptimisticConsensusState`  on malicious contracts. Since that takes a clientId rather than a lightClient address directly, the clientId must already be included in our mapping for the tx to go through.  However, I think we can still get this same effect with passing in the connection instead, and looking up the client that maps to this connection, so we're still only calling whitelisted light clients.

Reply:
I think having strings as the mapping should be fine; so using strings for now - should be added!